osu.Game/OsuGameBase.cs

```diff
@@ -356,6 +363,23 @@ List<ScoreInfo> getBeatmapScores(BeatmapSetInfo set)
             Ruleset.BindValueChanged(onRulesetChanged);
         }
 
+        private IDisposable blocking;
+
+        private void updateThreadStateChanged(ValueChangedEvent<GameThreadState> state)
+        {
+            switch (state.NewValue)
+            {
+                case GameThreadState.Running:
+                    blocking.Dispose();
```

Comment:
I'd probably throw in a null-check here, just for safety's sake:

```suggestion
                    blocking?.Dispose();
```